### PR TITLE
simple search feature working

### PR DESCRIPTION
**Pull request by Emmah-lou**
simple implementation of the search feature working well, need to have more filled out data in supabase (ect, size location, all the things to run the filters on the searched results)

### CODE REVIEW

**Comment by Emmah-lou** [comment]
going to bed now! see ya in the AM

File: src/components/shared/Header.jsx [hunk 1]
```diff
@@ -35,6 +36,8 @@ const Dropdown = () => {
 
 const Header = ({ toggleTheme }) => {
   const USER = useSelector((state) => state.auth);
+  const { searchQuery, handleInputChange, searchResults, searchTransActions } = useSearch();
+  const navigate = useNavigate();
```
Comment (BlocksOnChainCode) [inline src/components/shared/Header.jsx:40]: 👍 YepYep

File: src/pages/Search.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React, { useEffect } from 'react';
+import Layout from '../components/layout/Layout';
+import { useParams, Link } from 'react-router-dom';
+import useSearch from '../hooks/useSearch';
+import { TransAction } from '../components/feed/TransAction';
+const Search = () => {
+  const { query } = useParams();
```
Comment (BlocksOnChainCode) [inline src/pages/Search.jsx:7]: This could have gone in the hook, but no real need. All very clean 👍

File: src/hooks/useSearch.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  const searchTransActions = async (searchQuery) => {
+    const { data, error } = await supabase.from('transactions').select('*, profiles_id(username, avatar_url)').ilike('title', `%${searchQuery}%`);
+    if (error) {
+      console.log(error);
+    }
+    if (data.length === 0) {
+      console.log('No results found');
+    }
+    setSearchResults(data);
+  };
```
Comment (BlocksOnChainCode) [inline src/hooks/useSearch.js:20]: missing return keyword. (But I'm nitpicking! absolutely) wonderful work 😸  
Possibly a guard, I think in this case if data is empty, the previous will be removed and set to nothing still wouldn't it? 

`if (!data) {
    return; 
}`

**Review by BlocksOnChainCode — APPROVED** [review]
Good work 😄 

I am wondering however, the decision to have a search page and route, is due to the need to use useParams from the router I assume? 

I see no issues with that naturally, but I thought we was going for a component on top of feed for this? 

Also, give me your opinion on this review in general, when doing the nitpicking. Want me to merge or to send back? 

Now I will just merge this in 👍

**Pull request merged**